src/plugins/janus_audiobridge.c

```diff
@@ -8655,29 +8658,55 @@ static void *janus_audiobridge_participant_thread(void *data) {
 						janus_audiobridge_buffer_packet_destroy(bpkt);
 						break;
 					}
+
 					rtp = (janus_rtp_header *)bpkt->buffer;
 					/* If this is Opus, check if there's a packet gap we should fix with FEC */
 					use_fec = FALSE;
 					if(!first && participant->codec == JANUS_AUDIOCODEC_OPUS && participant->fec) {
-						if(ntohs(rtp->seq_number) == (participant->expected_seq + 1)) {
-							/* Lost a packet here? Use FEC to recover */
+						if(ntohs(rtp->seq_number) != (participant->expected_seq)) {
+							lost_packets = ntohs(rtp->seq_number) - participant->expected_seq;
+							JANUS_LOG(LOG_ERR, "[%d] got rtp seq, expected_seq: [%d], diff: [%d]\n", ntohs(rtp->seq_number), participant->expected_seq, lost_packets);
+
+							int32_t output_samples;
+							opus_decoder_ctl(participant->decoder, OPUS_GET_LAST_PACKET_DURATION(&output_samples));
+
+							for(int i=1; i < lost_packets; i++) {
+								pkt = g_malloc(sizeof(janus_audiobridge_rtp_relay_packet));
+								pkt->data = g_malloc0(OPUS_SAMPLES * (participant->stereo ? 2 : 1) * sizeof(opus_int16));
+								pkt->ssrc = 0;
+								pkt->timestamp = participant->last_timestamp + 960 * i;
+								pkt->seq_number = participant->last_seq + 1 * i;
```

Comment:
I'm not 100% sure the RTP info here is correct, since `last_timestamp` and `last_seq` may come from out of order packets. But it's probably not that relevant, since it has a limited use anyway in the code.

Reply:
I thought jitter buffer is reordering out of order packets and duplicates? At least I've expiremented with out of order and duplicates on output and everything still worked fine